config/initializers/opentelemetry.rb

```diff
@@ -0,0 +1,56 @@
+# config/initializer/opentelemetry.rb
+
+require "opentelemetry/sdk"
+require "opentelemetry/exporter/otlp"
+require "opentelemetry/instrumentation/all"
+
+class BatchSpanProcessorWithDbStatementFlattening < OpenTelemetry::SDK::Trace::Export::BatchSpanProcessor
```

Comment:
I don't like to subclass generally, but this works. Probably a better way to do this with delegation, but it's just a proof of concept.